homeassistant/components/enasolar/config_flow.py

```diff
@@ -0,0 +1,218 @@
+                if error is not None:
+                    _errors[CONF_HOST] = error
+                else:
+                    if self._conf_for_inverter_exists(self.get_serial_no()):
+                        return self.async_abort(reason="already_configured")
+                    await self.async_set_unique_id(self.get_serial_no())
+                    return await self.async_step_inverter()
+        else:
+            user_input = {}
+            user_input[CONF_NAME] = DEFAULT_NAME
+            user_input[CONF_HOST] = DEFAULT_HOST
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=user_input[CONF_HOST]): cv.string,
+                    vol.Optional(CONF_NAME, default=user_input[CONF_NAME]): cv.string,
+                }
+            ),
+            errors=_errors,
+            last_step=False,
+        )
+
+    async def async_step_inverter(self, user_input=None):
+        """Give the user the opportunities to override inverter config."""
+
+        _errors = {}
+        if user_input is not None:
+            # CAPABILITY is a 9 bit value, bits 1-3 and 9 representing
+            # what features the Innverter has.
+            if user_input[CONF_CAPABILITY] & ~CAPABILITY:
+                _errors[CONF_CAPABILITY] = "capability_invalid"
+            else:
+                self._data.update(user_input)
+                title = self._data[CONF_NAME]
+                return self.async_create_entry(title=title, data=self._data)
+        else:
+            # Use the capability bits from the Inverter. This assumes it
+            # had been possible to actually scrap them from the jScript
+            user_input = {}
+            user_input[CONF_MAX_OUTPUT] = self._enasolar.get_max_output()
+            user_input[CONF_DC_STRINGS] = self._enasolar.get_dc_strings()
+            user_input[CONF_CAPABILITY] = self._enasolar.get_capability()
+
+        return self.async_show_form(
+            step_id="inverter",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_MAX_OUTPUT, default=user_input[CONF_MAX_OUTPUT]
+                    ): vol.In(MAX_OUTPUT),
+                    vol.Required(
+                        CONF_DC_STRINGS, default=user_input[CONF_DC_STRINGS]
+                    ): vol.In(DC_STRINGS),
+                    vol.Required(
+                        CONF_CAPABILITY, default=user_input[CONF_CAPABILITY]
+                    ): vol.Coerce(int),
+                }
+            ),
+            errors=_errors,
+            last_step=True,
+        )
+
+
+class EnaSolarOptionsFlowHandler(config_entries.OptionsFlow):
+    """Allow Polling window to be updated."""
+
+    def __init__(self, config_entry):
+        """Initialize EnaSolar options flow."""
+        self.config_entry = config_entry
+        self._options = dict(config_entry.options)
```

Comment:
```suggestion
        self._options = config_entry.options
```
Should already be a dict?

Reply:
Yep, updated

Reply:
If I remove this, the the update method isn't available for _options.